src/sdk/Recognizer.ts

```diff
@@ -186,7 +186,7 @@ export abstract class Recognizer {
                 new Context(new OS(osPlatform, osName, osVersion))));
 
         const subscriptionKey = this.privProperties.getProperty(PropertyId.SpeechServiceConnection_Key, undefined);
-        const authentication = subscriptionKey ?
+        const authentication = (subscriptionKey && subscriptionKey !== "") ?
             new CognitiveSubscriptionKeyAuthentication(subscriptionKey) :
             new CognitiveTokenAuthentication(
                 (authFetchEventId: string): Promise<string> => {
```

Comment:
just curios: why do we need to do this twice?

Reply:
yes, the wonder of js comparism rules

Reply:
that is from the previous lib implementation where there was a separation of first and subsequent retrievals

---
In reply to: [258072439](https://github.com/Microsoft/cognitive-services-speech-sdk-js/pull/36#discussion_r258072439) [](ancestors = 258072439)